quantum/pointing_device/pointing_device_auto_mouse.c

```diff
@@ -0,0 +1,207 @@
+/* Copyright 2021 Christopher Courtney, aka Drashna Jael're  (@drashna) <drashna@live.com>
+ * Copyright 2022 Alabastard
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "pointing_device_auto_mouse.h"
+
+/* needed variables for tracking auto mouse */
+static auto_mouse_context_t auto_mouse = {.config.layer = (uint8_t)AUTO_MOUSE_DEFAULT_LAYER};
+
+/* Handle checking if a keycode is a mouse button */
+static bool is_mouse_record(uint16_t keycode, keyrecord_t* record) {
+    // allow for keyboard to hook in and override if need be
+    if (is_mouse_record_kb(keycode, record)) return true;
+    switch (keycode) {
+        case KC_MS_UP ... KC_MS_WH_RIGHT:
+            return true;
+    }
+    return false;
+}
+
+/* toggle mouse layer setting (disabling automouse layer changes) */
+void toggle_mouse_layer(void) {
+    auto_mouse.status.layer_toggled ^= 1;
+}
+
+/* Allow for getting auto mouse state */
+bool get_auto_mouse_state(void) {
+    return auto_mouse.config.enabled;
+}
+
+/* Allow for setting auto mouse enable state */
+void set_auto_mouse_state(bool state) {
+    // skip if already set
+    if (auto_mouse.config.enabled == state) return;
+    auto_mouse.config.enabled = state;
+    // reset settings on state change
+    auto_mouse.status.mouse_key_tracker = 0;
+    auto_mouse.timer.active             = 0;
+    auto_mouse.timer.delay              = 0;
+    auto_mouse.status.layer_toggled     = false;
```

Comment:
This is more of a nit.
```suggestion
    if (auto_mouse.config.enabled != state) {
        auto_mouse.config.enabled = state;
        // reset settings on state change
        auto_mouse.status.mouse_key_tracker = 0;
        auto_mouse.timer.active             = 0;
        auto_mouse.timer.delay              = 0;
        auto_mouse.status.layer_toggled     = false;
    }
```